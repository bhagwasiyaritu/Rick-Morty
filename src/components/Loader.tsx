import {ActivityIndicator} from 'react-native';
import React from 'react';

const Loader = () => {
  return <ActivityIndicator testID='loader' size="large" className="color-violet-900" />;
};

export default Loader;
