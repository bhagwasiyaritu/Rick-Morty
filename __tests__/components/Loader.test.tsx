import React from 'react';
import {render} from '@testing-library/react-native';
import Loader from '../../src/components/Loader';

describe('Loader', () => {
  it('renders correctly', () => {
    const {getByTestId} = render(<Loader />);
    const loader = getByTestId('loader');

    expect(loader).toBeTruthy();
  });

  it('matches snapshot', () => {
    const {toJSON} = render(<Loader />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('has correct size and color', () => {
    const {getByTestId} = render(<Loader />);
    const loader = getByTestId('loader');
    
    expect(loader.props.size).toBe('large');
    expect(loader.props.className).toBe('color-violet-900');
  });
});
