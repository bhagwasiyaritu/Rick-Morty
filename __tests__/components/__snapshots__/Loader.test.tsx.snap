// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot 1`] = `
<ActivityIndicator
  className="color-violet-900"
  size="large"
  testID="loader"
/>
`;
